packages/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java

```diff
@@ -129,16 +129,32 @@ private Source getSource(Map<String, Object> arguments) {
     List<Object> data = new ArrayList<>();
     if (orderBy != null) {
       for (List<Object> order : orderBy) {
-        String orderByFieldName = (String) order.get(0);
-        if (orderByFieldName.contains(".")) {
-          String[] fieldNameParts = orderByFieldName.split("\\.");
-          Map<String, Object> current = (Map<String, Object>) documentData.get(fieldNameParts[0]);
-          for (int i = 1; i < fieldNameParts.length - 1; i++) {
-            current = (Map<String, Object>) current.get(fieldNameParts[i]);
+        final Object field = order.get(0);
+
+        if (field instanceof FieldPath) {
+          if (field == FieldPath.documentId()) {
+            // This is also checked by an assertion on the Dart side.
+            throw new IllegalArgumentException(
+                "You cannot order by the document id when using"
+                    + "{start/end}{At/After/Before}Document a the library will order by the document id"
```

Comment:
It looks like there is some typo here, mind to check?

Reply:
thanks 👍